src/lib_ccx/networking.c

```diff
@@ -335,7 +335,7 @@ int net_udp_read(int socket, void *buffer, size_t length, const char *src_str, c
 	assert(length > 0);
 
 	int i;
-	char ip[50] = "";
+	char ip[15];
```

Comment:
I'd say we are one short here (3*4 for digits + 3 for dots + final zero)
Also it would be good to have that as a constant rather than appear twice as a magic number

Reply:
Should I fix the other arrays as well? https://github.com/jshin313/ccextractor/blob/594a83cc4ec76b61ebe2f81dbd6af5c090f0f57b/src/lib_ccx/networking.c#L1096

Reply:
If you want to create a PR for this today, yes, please :)

Reply:
Should I use the INET_ADDRSTRLEN constant? The <netinet/in.h> POSIX®header file says the constant is 16 bytes. But the Windows version is 22 bytes
`// Maximum length of address literals (potentially including a port number)
// generated by any address-to-string conversion routine.  This length can
// be used when declaring buffers used with getnameinfo, WSAAddressToString,
// inet_ntoa, etc.  We just provide one define, rather than one per api,
// to avoid confusion.
//
// The totals are derived from the following data:
//  15: IPv4 address
//  45: IPv6 address including embedded IPv4 address
//  11: Scope Id
//   2: Brackets around IPv6 address when port is present
//   6: Port (including colon)
//   1: Terminating null byte
//
#define INET_ADDRSTRLEN  22`

Reply:
Seems like a good idea to me to use a built-in constant :)

Reply:
If there is a portable built-in constant it's definitely best to use it.
If there isn't one define your own, but really there's nothing wrong with being generous - a few bytes wasted is much much better than an unexpected buffer overrun. Just add a comment next to the definition so the reader doesn't have to figure out where a magic number comes from.